src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/LicenseScanTests.cs

```diff
@@ -0,0 +1,243 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.SourceBuild.SmokeTests;
+
+[Trait("Category", "LicenseScan")]
+public class LicenseScanTests : TestBase
+{
+    private static readonly string[] s_allowedLicenseExpressions = new string[]
+    {
+        "apache-1.1", // https://opensource.org/license/apache-1-1/
+        "apache-2.0", // https://opensource.org/license/apache-2-0/
+        "apache-2.0 WITH llvm-exception", // https://foundation.llvm.org/relicensing/LICENSE.txt
+        "apsl-2.0", // https://opensource.org/license/apsl-2-0-php/
+        "bsd-new", // https://opensource.org/license/BSD-3-clause/
+        "bsd-original", // https://github.com/nexB/scancode-toolkit/blob/develop/src/licensedcode/data/licenses/bsd-original.LICENSE
+        "bsd-original-uc", // https://github.com/nexB/scancode-toolkit/blob/develop/src/licensedcode/data/licenses/bsd-original-uc.LICENSE
+        "bsd-simplified", // https://opensource.org/license/bsd-2-clause/
+        "bytemark", // https://github.com/nexB/scancode-toolkit/blob/develop/src/licensedcode/data/licenses/bytemark.LICENSE
+        "bzip2-libbzip-2010", // https://github.com/nexB/scancode-toolkit/blob/develop/src/licensedcode/data/licenses/bzip2-libbzip-2010.LICENSE
+        "cc0-1.0", // https://creativecommons.org/publicdomain/zero/1.0/legalcode
+        "cc-by-sa-3.0", // https://creativecommons.org/licenses/by-sa/3.0/legalcode
+        "cc-by-sa-4.0", // https://creativecommons.org/licenses/by-sa/4.0/legalcode
+        "cc-pd", // https://creativecommons.org/publicdomain/mark/1.0/
+        "cecill-c", // https://cecill.info/licences/Licence_CeCILL-C_V1-en.txt
+        "epl-1.0", // https://opensource.org/license/epl-1-0/
+        "generic-cla", // https://github.com/nexB/scancode-toolkit/blob/develop/src/licensedcode/data/licenses/generic-cla.LICENSE
+        "gpl-1.0-plus", // https://opensource.org/license/gpl-1-0/
+        "gpl-2.0", // https://opensource.org/license/gpl-2-0/
+        "gpl-2.0-plus WITH autoconf-simple-exception-2.0", // https://github.com/nexB/scancode-toolkit/blob/develop/src/licensedcode/data/rules/gpl-2.0-plus_with_autoconf-simple-exception-2.0_8.RULE
+        "gpl-2.0 WITH gcc-linking-exception-2.0", // https://github.com/nexB/scancode-toolkit/blob/develop/src/licensedcode/data/rules/gpl-2.0_with_gcc-linking-exception-2.0_6.RULE
+        "isc", // https://opensource.org/license/isc-license-txt/
+        "json", // https://www.json.org/license.html
```

Comment:
Do you know what matches this license? From the attached logs, it seems to be `src/core-sdk-tasks/BuildFPMToolPreReqs.cs`, which seems like a false positive?

The JSON License is often considered a non-open source license:

- Debian: https://wiki.debian.org/qa.debian.org/jsonevil
- Fedora: https://docs.fedoraproject.org/en-US/legal/not-allowed-licenses/
- FSF/GNU: https://www.gnu.org/licenses/license-list.html#JSON
- IBM: https://www.youtube.com/watch?v=-C-JoyNuQJs&t=2480s

IMO, we shouldn't allow this license.

Reply:
It's finding `json` license references in the following locations:

* https://github.com/dotnet/installer/blob/d7ad055f7ff0f04ff95d54b5cd060a751cf0f14b/src/core-sdk-tasks/BuildFPMToolPreReqs.cs#L111-L112
* https://github.com/dotnet/command-line-api/blob/fa5def68df3be879b1d73c2edbb7a6fbf5afec26/System.CommandLine.sln#L16-L17
* https://github.com/dotnet/arcade/blob/60a2e897fb13b2042e64d3be6a4f9d43f39b6f07/src/Microsoft.DotNet.Build.Tasks.Installers/src/BuildFPMToolPreReqs.cs#L138-L139
* https://github.com/xunit/xunit/blob/f110e5bee5dfd4c08339587c9c3df9292fcb597c/xunit.sln#L20-L21
* https://github.com/NuGet/NuGet.Client/blob/681bf33f7d16747a3ce699f016739e8bf6281c69/test/TestExtensions/GenerateLicenseList/Program.cs#L26
* https://github.com/dotnet/runtime/blob/0cc9f2193bb447db60a81b05986ef4f513bd2381/src/libraries/System.Configuration.ConfigurationManager/tests/Mono/LongValidatorTest.cs#L2-L28

So yes, these all appear to be false positives. The last one is especially perplexing as there is no mention of "json" or "evil". I'll log scancode issues as appropriate to address false positives. And I'll remove json as an allowed license. Thanks for educating me!

Reply:
Ah. I just reported https://github.com/nexB/scancode-toolkit/issues/3531 that covers the xunit.sln file.

Reply:
Fixed in https://github.com/dotnet/installer/pull/17442/commits/a7306a4de65bed34812587cc9b103c4644529987